Dockerfile

```diff
@@ -1,100 +1,97 @@
+# clone ros package repo
+ENV NAV2_WS /opt/nav2_ws
+RUN mkdir -p $NAV2_WS/src
+WORKDIR $NAV2_WS/src
+ARG GIT_REPO_URL=https://github.com/ros-planning/navigation2.git
+RUN git clone $GIT_REPO_URL navigation2
 
 # change to correct branch if $BRANCH is not = master
-WORKDIR /ros2_ws/navigation2_ws/src/navigation2
 ARG PULLREQ=false
 ARG BRANCH=master
 RUN echo "pullreq is $PULLREQ"
-RUN if [ "$PULLREQ" == "false" ] && [ "$BRANCH" == "master" ]; then \ 
+RUN if [ "$PULLREQ" = "false" ] && [ "$BRANCH" = "master" ]; then \
       echo "No pull request number given - defaulting to master branch"; \
     elif [ "$BRANCH" != "master" ]; then \
       cd navigation2; \
       git fetch origin $BRANCH:temp_branch; \
       git checkout temp_branch; \
-      cd -; \
       echo "No pull request number given - defaulting to $BRANCH branch"; \
     else \
+      cd navigation2; \
       git fetch origin pull/$PULLREQ/head:pr_branch; \
       git checkout pr_branch; \
     fi
 
-# Download dependencies
-RUN echo "Downloading the ROS 2 navstack dependencies workspace"
-WORKDIR /ros2_ws/navstack_dependencies_ws/src
-RUN vcs import . < /ros2_ws/navigation2_ws/src/navigation2/tools/ros2_dependencies.repos
+# clone dependency package repos
+ENV ROS_WS /opt/ros_ws
+RUN mkdir -p $ROS_WS/src
+WORKDIR $ROS_WS
+RUN vcs import src < $NAV2_WS/src/navigation2/tools/ros2_dependencies.repos
+
+# install dependency package dependencies
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    apt-get update && \
+    rosdep install -q -y \
+      --from-paths \
+        src \
+      --ignore-src \
+    && rm -rf /var/lib/apt/lists/*
 
-# Build ROS 2 dependencies
-WORKDIR /ros2_ws/navstack_dependencies_ws
-RUN rosdep init && rosdep update && rosdep install -q -y -r --from-paths src --ignore-src --rosdistro $ROS2_DISTRO --as-root=apt:false --as-root=pip:false --skip-keys "catkin"
-# source
-RUN (source /opt/ros/$ROS2_DISTRO/setup.bash && . /ros2_ws/ros2-linux/setup.bash && colcon build --symlink-install)
+# build dependency package source
+ARG CMAKE_BUILD_TYPE=Release
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    colcon build \
+      --symlink-install \
+      --cmake-args \
+        -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE
 
-# Build navigation2 code
-WORKDIR /ros2_ws/navigation2_ws
-RUN rm /ros2_ws/navigation2_ws/src/navigation2/nav2_system_tests/COLCON_IGNORE
-RUN rosdep install -q -y -r --from-paths src --ignore-src --rosdistro $ROS2_DISTRO --as-root=apt:false --as-root=pip:false
-RUN (. /ros2_ws/navstack_dependencies_ws/install/setup.bash && colcon build --symlink-install)
+# install navigation2 package dependencies
+WORKDIR $NAV2_WS
+RUN . $ROS_WS/install/setup.sh && \
+    apt-get update && \
+    rosdep install -q -y \
+      --from-paths \
+        $ROS_WS/src \
+        src \
+      --ignore-src \
+    && rm -rf /var/lib/apt/lists/*
 
-RUN echo "export ROS_DOMAIN_ID=22" >> /root/.bashrc
-COPY tools/ctest_retry.bash /ros2_ws/navigation2_ws/build/nav2_system_tests
```

Comment:
We didn't merge #527 because we found a real issue that is being addressed in #538, so we DO want the system tests to stay enabled. If you have a better way to do this from the Travis file, please add it to the PR.

Reply:
What exact command do you want to run?  
We can just add it to the current order of scripts in the .travis.yaml

Reply:
The Travis CI is calling the ctest_retry, which runs the system tests and retries up to 3 times, if they pass once, it returns true. This is because we have some system issues that cause about a 20% flaky test rate. We're working on fixing those by changing to use lifecycle nodes in the future, but this allows us to get the flaky fail rate down to ~1% and enable the test in CI to validate PRs. We don't want to remove this without another equivalent method.